### PR TITLE
Output environment file for Windows

### PR DESCRIPTION
**Pull request by edoardopirovano**
Currently, the environment file for the tracer is only being output in the Unix format. Thus, these variables are not being read by the tracer on Windows machines. This PR addresses this by also outputting the environment file in the Windows format.

I have also modified the `runner-analyze-csharp-windows` test to unset the `CODEQL_EXTRACTOR_CSHARP_ROOT` variable before the build step. This would cause the test to fail if the environment file being produced was not correct.

### CODE REVIEW

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -465,7 +465,12 @@ jobs:
 
     - name: Build code
       shell: powershell
+      # Note this step unsets the CODEQL_EXTRACTOR_CSHARP_ROOT to ensure that the .win32env file is read correctly
```
[the remark this answers is not in the document]
Comment (edoardopirovano) [reply .github/workflows/pr-checks.yml:468]: Before this PR the file was not being produced at all, so I figured we need to modify an integration test to make sure it actually gets read when it is produced.

[the remark this answers is not in the document]
Comment (aeisenberg) [reply .github/workflows/pr-checks.yml:468]: Oh. I see, I misunderstood the comment and what you're doing here. By un-setting the `CODEQL_EXTRACTOR_CSHARP_ROOT` variable, you are ensuring that `.win32env` is being read correctly. 

Could you update the comment to something like this?

```suggestion
      # Note we want to make sure that the .win32env file is read correctlyl, so we unset the CODEQL_EXTRACTOR_CSHARP_ROOT from the .sh file.
```

**Review by aeisenberg — APPROVED** [review]

File: .github/workflows/pr-checks.yml [hunk 2]
```diff
@@ -465,7 +465,12 @@ jobs:
...
       run: |
+        (Get-Content ./codeql-runner/codeql-env.sh) | Select-String -pattern ".*CODEQL_EXTRACTOR_CSHARP_ROOT.*" -NotMatch | ? {$_.ToString().trim() -ne "" } | Set-Content ./codeql-runner/codeql-env.sh
+        $jsonEnv = Get-Content -Path ./codeql-runner/codeql-env.json -Raw | ConvertFrom-Json
+        $jsonEnv.PSObject.Properties.Remove('CODEQL_EXTRACTOR_CSHARP_ROOT')
+        $jsonEnv | ConvertTo-Json -Depth 1 | Out-File ./codeql-runner/codeql-env.json -Force
```
Comment (adityasharad) [inline .github/workflows/pr-checks.yml:473]: Could we make this simpler, by removing `CODEQL_EXTRACTOR_CSHARP_ROOT` from the environment after the `Invoke-Expression` step below? I think `Clear-Variable` is the PowerShell command.
The step below doesn't use the JSON file.

File: src/tracer-config.ts [hunk 3]
```diff
@@ -138,10 +138,30 @@ export function concatTracerConfigs(
     sizeBuffer.writeInt32LE(lineBuffer.length, 0);
     buffer = Buffer.concat([buffer, sizeBuffer, lineBuffer]);
   }
-  // Write the compound environment
+  // Write the compound environment for Unix
   const envPath = `${spec}.environment`;
   fs.writeFileSync(envPath, buffer);
 
+  // Prepare the content of the compound environment file on Windows
```
Comment (adityasharad) [inline src/tracer-config.ts:145]: What do you think about doing this step only when we're on Windows, vs unconditionally?

File: .github/workflows/pr-checks.yml (hunk 2 above)
Comment (hmakholm) [reply .github/workflows/pr-checks.yml:473]: I would have a (weak!) preference to only write the file that corresponds to the current platform. I doubt it makes any _concrete_ difference either way, but it feels most right to try to mimic what the actual `preload_tracer` does.

File: src/tracer-config.ts (hunk 3 above)
Comment (hmakholm) [reply src/tracer-config.ts:145]: I would have a (weak!) preference to only write the file that corresponds to the current platform. I doubt it makes any _concrete_ difference either way, but it feels most right to try to mimic what the actual `preload_tracer` does.

File: .github/workflows/pr-checks.yml (hunk 2 above)
Comment (edoardopirovano) [reply .github/workflows/pr-checks.yml]: Aha, done. For reason I thought those files needed patching to stop the runner reading them again.

File: src/tracer-config.ts (hunk 3 above)
Comment (edoardopirovano) [reply src/tracer-config.ts]: Sure, done - except in the unit tests where we still test both regardless of platform.

File: .github/workflows/pr-checks.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (edoardopirovano) [reply .github/workflows/pr-checks.yml]: Sure, done ✅

**Pull request merged**